app/src/pages/CraftableTokenPage.js

```diff
@@ -110,19 +108,25 @@ class CraftableTokenPage extends React.Component {
     try {
       await token.approve(
         this.props.store.domain.crafty.address,
-        new BN(change.newValue ? '57896044618658097711785492504343953926634992332820282019728792003956564819968' : '0', 10) // 2*255
+        (new BN(2))
+          .pow(new BN(256))
+          .sub(new BN(1))
+        // ^ 2^256 - 1
       )
 
       runInAction(() => {
         this.expectingChange.set(address, true)
       })
       // set local pending
     } catch (error) {
+      // @TODO(shrugs) - notify user of error
+      console.error(error)
+    } finally {
+      await when(() => field.$('approved').values())
       runInAction(() => {
         this.expectingChange.set(address, false)
       })
-      // cancel local pending
-      console.error(error)
+      field.$('pending').set(false)
```

Comment:
What about rejections or TX failures? The `when` condition will never be true, and `pending` won't be set back to false

Reply:
The whole `finally` block is a bit weird, since it only makes sense for the successful case

Reply:
hmm yeah. the reason that `when` is there is to stop the UI from flashing in between `request finishes -> pending set to false, approval still false, ui updates -> mobx reloads the allowances -> ui updates to show approval granted`.

Reply:
I see. I moved the `when` to the `try` block and rejections are now handled properly. We still get some flashing though, but I think it's due to how everything is updated from scratch every 15s